### PR TITLE
Split forms for reuse

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

**Comment by huwshimi** [comment]
QA:
- the payment forms should appear and function as they did before (see: https://github.com/juju/juju-gui/pull/2700#issuecomment-290592474)

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7078/

File: jujugui/static/gui/src/app/components/card-form/card-form.js [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+
+    /**
+      Handle updating the number with the correct spacing.
+
+      @method _handleNumberChange
+    */
+    _handleNumberChange: function() {
+      this.refs.number.setValue(
+        this._formatNumber(this.refs.number.getValue()));
+      // TODO: when we have card logos we should update the logo for the card
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/card-form/card-form.js]: We should have a task somewhere for this, TODO's have a knack for getting lost.

File: jujugui/static/gui/src/app/components/card-form/card-form.js [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        numbers: /^4/,
+        blocks: [4, 4, 4, 4]
+      }, {
+        name: 'MasterCard',
+        // Prefix: 50-55
+        numbers: /^5[1-5]/,
+        blocks: [4, 4, 4, 4]
+      }, {
+        name: 'Discover',
+        // Prefix: 	6011, 622126-622925, 644-649, 65
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/card-form/card-form.js]: some extra spaces here :)

File: jujugui/static/gui/src/app/assets/css/_components.scss [hunk 3]
```diff
@@ -5,11 +5,13 @@
 @import '../../components/account/payment-methods/payment-methods';
 @import '../../components/account/payment-methods/card/card';
 @import '../../components/added-services-list/added-services-list';
+@import '../../components/address-form/address-form';
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/css/_components.scss]: Yep, I'll be adding code in there shortly.

File: jujugui/static/gui/src/app/assets/css/_components.scss [hunk 4]
```diff
@@ -5,11 +5,13 @@
...
 @import '../../components/boolean-config/boolean-config';
 @import '../../components/budget-chart/budget-chart';
 @import '../../components/budget-table/budget-table';
 @import '../../components/budget-table/row/row';
 @import '../../components/button-row/button-row';
+@import '../../components/card-form/card-form';
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/css/_components.scss]: To be used very soon.

File: jujugui/static/gui/src/app/components/card-form/card-form.js (hunk 1 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/card-form/card-form.js]: I'll add a card. Thanks.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/juju-gui/7084/

**Comment by huwshimi** [comment]
Thanks @hatched :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**